### PR TITLE
refactor: create Service Package

### PR DESCRIPTION
**Pull request by obinns-dd**
# refactor: create Service Package

**Implements:** [DCMAW-10076](https://govukverify.atlassian.net/browse/DCMAW-10076)

Network requests to the backend (such as hello world, appInfo, etc.) will be encapsulated within Service classes in a new `MobilePlatformServices` module, similar to the pattern used by the ID Check app codebase.

This means that we can extract the details of API calls out of any coordinators / UI logic, and keep the One Login app codebase cleaner.

- [x] Rebase following the merge of #185 

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update
- [x] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [ ] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [ ] ~~Met all accessibility requirements?~~ - **No functional change in this pull request**
    - [ ] Checked dynamic type sizes are applied
    - [ ] Checked VoiceOver can navigate your new code
    - [ ] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.

[DCMAW-10076]: https://govukverify.atlassian.net/browse/DCMAW-10076?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: MobilePlatformServices/Sources/MobilePlatformServices/App Info/AppInformationProvider.swift [hunk 1]
```diff
@@ -28,7 +29,7 @@ public final class AppInformationService: AppInformationServicing {
     }
     
     public func fetchAppInfo() async throws -> App {
-        var request = URLRequest(url: AppEnvironment.appInfoURL)
```
Comment (Magdaline-A) [inline MobilePlatformServices/Sources/MobilePlatformServices/App Info/AppInformationProvider.swift:31]: Should we remove this property from the `AppEnvironment` file now that we're not using it?

**Review by Magdaline-A — APPROVED** [review]

File: MobilePlatformServices/Package.resolved [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      "identity" : "mobile-ios-networking",
+      "kind" : "remoteSourceControl",
+      "location" : "https://github.com/govuk-one-login/mobile-ios-networking",
+      "state" : {
+        "revision" : "7ea8b9091042545403eb1db2ad2cc60e9ce17191",
+        "version" : "3.0.0"
+      }
+    }
+  ],
+  "version" : 3
```
Comment (mcgarj) [inline MobilePlatformServices/Package.resolved:14]: Should this be version 2 and the `originHash` be removed? I noticed Xcode loves to add this when branch changing for some reason - or does this not overly matter for a package?

Comment (obinns-dd) [reply MobilePlatformServices/Package.resolved:14]: The latest version of Xcode using version 3 of this file 🙂 previously this caused issues on the CI which had an older version of Xcode. This shouldn't be a problem anymore: https://github.com/govuk-one-login/mobile-ios-one-login-app/pull/190

**Review by mcgarj — APPROVED** [review]
🙌

**Pull request merged**